build.gradle.kts

```diff
@@ -0,0 +1,306 @@
+import com.github.benmanes.gradle.versions.updates.DependencyUpdatesTask
+
+import io.gitlab.arturbosch.detekt.detekt
+
+import org.gradle.api.tasks.testing.logging.TestExceptionFormat
+import org.gradle.api.tasks.testing.logging.TestLogEvent
+import org.gradle.plugins.ide.idea.model.IdeaModel
+import org.gradle.plugins.ide.idea.model.IdeaProject
+
+import org.jetbrains.dokka.gradle.DokkaTask
+import org.jetbrains.gradle.ext.DefaultRunConfigurationContainer
+import org.jetbrains.gradle.ext.JUnit
+import org.jetbrains.gradle.ext.ProjectSettings
+import org.jetbrains.gradle.ext.RunConfiguration
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+import org.jetbrains.kotlin.gradle.plugin.KotlinSourceSet
+
+import java.io.ByteArrayOutputStream
+import java.net.URL
+
+val detektPluginVersion: String by project
+val kotlinPluginVersion: String by project
+
+val kotlintestVersion: String by project
+
+plugins {
+    kotlin("jvm") apply false
+
+    id("io.gitlab.arturbosch.detekt") apply false
+    id("org.jetbrains.dokka") apply false
+
+    id("com.github.ben-manes.versions")
+    id("org.ajoberstar.grgit")
+    id("org.jetbrains.gradle.plugin.idea-ext")
+}
+
+// The following extension functions add the missing Kotlin DSL syntactic sugar for configuring the idea-ext plugin.
+
+fun Project.idea(block: IdeaModel.() -> Unit) =
+    (this as ExtensionAware).extensions.configure("idea", block)
+
+fun IdeaProject.settings(block: ProjectSettings.() -> Unit) =
+    (this@settings as ExtensionAware).extensions.configure(block)
+
+fun ProjectSettings.runConfigurations(block: DefaultRunConfigurationContainer.() -> Unit) =
+    (this@runConfigurations as ExtensionAware).extensions.configure("runConfigurations", block)
```

Comment:
IIRC you use similar extensions in another build file, could they go to `buildSrc`?

Reply:
Indeed that was my plan, but I couldn't get it to work. I'll try again as a post-merge clean-up step.

Reply:
See https://github.com/heremaps/oss-review-toolkit/pull/1468.